src/applications/vaos/project-cheetah/redux/actions.js

```diff
@@ -132,6 +141,51 @@ export function getClinics({ facilityId, showModal = false }) {
   };
 }
 
+export function openNewBookingPage(history) {
+  return async (dispatch, getState) => {
+    dispatch({
+      type: FETCH_NEW_BOOKING_WINDOW,
+    });
+
+    let isEligible = false;
+    try {
+      const initialState = getState();
+      const siteIds = selectSystemIds(initialState);
+
+      // Get sites that support vaccines
+      const criteria = await getDirectBookingEligibilityCriteria(siteIds);
+      isEligible = criteria.some(setting =>
+        setting.coreSettings.some(
+          coreSetting =>
+            coreSetting.id === TYPE_OF_CARE_ID &&
+            !!coreSetting.patientHistoryRequired,
+        ),
+      );
+
+      // Redirect the user to the 'Contact facility' page if the appointment can't be
+      // scheduled at the user's registered facilities.
+      if (!isEligible) {
+        history.push('/new-project-cheetah-booking/contact-facilities');
+      }
+      // TODO: Remove!!!
+      // history.push('/new-project-cheetah-booking/contact-facilities');
+      dispatch({
+        type: FETCH_NEW_BOOKING_WINDOW_SUCCEEDED,
+        isEligible,
+      });
+    } catch (e) {
+      dispatch({
+        type: FETCH_NEW_BOOKING_WINDOW_FAILED,
+        isEligible,
+      });
+
+      // Just capture the error for now.
+      // TODO: Figure out where to redirect the user.
+      captureError(e, false);
```

Comment:
Jeff mentioned opening a new ticket this.

Reply:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/20203